eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/NatsController.kt

```diff
@@ -36,22 +37,21 @@ interface NatsController<ReqT : GeneratedMessageV3, RepT : GeneratedMessageV3> {
      */
     val subject: String
 
+    /**
+     * The handler responsible for processing the incoming NATS messages. It defines how the request messages
+     * of type [ReqT] should be handled and processed to produce a response of type [RepT].
+     * Any class implementing this interface should provide
+     * an instance of a subtype of [Handler] that corresponds to the types [ReqT] and [RepT].
+     */
+    val handler: H
+
     /**
      * Function to handle an incoming request message. Implementations should process the request
      * and return a corresponding reply message wrapped in a Mono for reactive handling.
      *
-     * @param message The incoming request message of type ReqT.
+     * @param request The incoming request message of type ReqT.
      * @return A reactive publisher (Mono) that emits a single reply message of type RepT.
      */
-    fun handler(request: ReqT): Mono<RepT>
+    fun handle(request: Mono<ReqT>): Mono<RepT>
```

Comment:
Why the request has been changed to `Mono<ReqT>`?

Reply:
I reduced code duplication in my NATS controllers and gRPC service.
Both were using the same code to handle protobuf messages. 
I moved the shared code to handlers. 
To ensure compatibility with both NATS and gRPC, I made it Mono

Reply:
Maybe you mean it was better to make **Mono** directly from **request: ReqT** at method **handle**?

Reply:
I opened this discussion because I see no reason to make the request parameter reactive.

I bet somewhere at client side of this code we can find:
```kotlin
controller. handle(Mono.just(request))
```

Do you know why passing optional as function parameter is bad idea?
```java
void method(Optional<String> arg) 
```

Similar case for `Mono` as argument.

Reply:
Now I understand what's wrong.
Fixing it

Reply:
`+`